transforms/no-implicit-this/__testfixtures__/dont-assume-this.options.json

```diff
@@ -1,3 +0,0 @@
-{
-  "dontAssumeThis": true
```

Comment:
why remove this behavior/test?

Reply:
a path can only be either:

- a property
- a scoped variable
- a component invocation
- a helper invocation

if it is not a component/helper invocation and not a scoped variable then there is only "property" left. if we use `dontAssumeThis: true` in what case would we then prefix the path with `this.`?

Reply:
good point. that would be confusing. I think I'll have to do a major bump, just in case though.

Reply:
> I think I'll have to do a major bump, just in case though.

yeah, totally agree. I'd wait a bit though so that people can try it out from `master`, and I might have a few more PRs coming in... 😅

Reply:
> I'd wait a bit though

too late!
I have no problem with multiple releases or multiple breaking releases even. 
release-it is my favorite tool for this stuff. so easy.

Reply:
So we were actually using this functionality 😬 but if it's not deemed useful then 🔪  

For our use case, while we know that passed in args are reflected onto a classic component and all its properties with can be referenced using `this.`, we actually didn't want to do just migrate all properties in our templates to `this.`. We wanted to try and semantically migrate our templates to properly disambiguate between `this.` vs `@` properties.

Our process for this was 

1. Run this codemod on said template
2. Turn on `no-implicit-this` ember-template-lint rule
3. Any properties which didn't have `this.` prepended to them we'd manually fix on a case-by-case basis depending on whether the property existing in the JS (i.e. `this.set('someProperty', ...)`) or just passed directly in.

Reply:
I think that's def a valid workflow, and should maybe be called out in the readme on this repo. @patocallaghan (or really anyone), a PR to add don't-assume-this back in would be worthwhile.

however, I keep forgetting, if you don't-assume-this, does the codemod do anything?

Reply:
no, in that case the codemod would literally do nothing, so I'm not sure if it's actually that useful 🤔

Reply:
So when we say it literally did nothing, before the default behaviour was _always_ to append `this.` but with this option it just opted you out of that left the property as is if it didn't find it in the component's telemetry. It was then on you to manually figure out if it was `this.` or `@`.

Reply:
that makes sense. We've had some issues with getting property information in telemetry from all the different types of classes / ember versions / ways of constructing "Meta". :(